packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -104,6 +138,71 @@ const modifiers = [
   },
 ];
 
+const ListButton = ({ icon, format, label }) => {
+  const editor = useSlate();
+
+  /**
+   *
+   * @param {import('slate').Node} node
+   * @returns boolean
+   */
+  const isListNode = (node) => {
+    return !Editor.isEditor(node) && Element.isElement(node) && node.type === 'list';
+  };
+
+  const isListActive = () => {
+    const { selection } = editor;
+
+    if (!selection) return false;
+
+    const [match] = Array.from(
+      Editor.nodes(editor, {
+        at: Editor.unhangRange(editor, selection),
+        match: (node) => isListNode(node) && node.format === format,
+      })
+    );
+
+    return Boolean(match);
+  };
+
+  const isActive = isListActive();
+
+  const toggleList = () => {
+    Transforms.unwrapNodes(editor, {
+      match: (node) => isListNode(node) && ['ordered', 'unordered'].includes(node.format),
+      split: true,
+    });
+
+    Transforms.setNodes(editor, {
+      type: isActive ? 'paragraph' : 'list-item',
```

Comment:
Yes if you toggle coming from a heading for example it would do heading => list item => paragraph => list item => paragraph

The other rich text editors I've tested have that behavior too